app/features/Type/Notes.py

```diff
@@ -33,18 +37,18 @@ def append(self, note, currentTime):
     # 2: middle
     # 3: middle end
     # 4: end
-    def getAverageByPartsTime(self):
+    def getAverageByPartsTime(self, nbSplit):
         averageMIDI = []
         j = 0
-        timeSplit = self.totalTime / 5
+        timeSplit = self.totalTime / nbSplit
         currentTimeSplit = timeSplit
         average = 0
         for i in range(len(self.notes)):
             average = average + self.notes[i]['note']
-            j = j + 1
-            if self.notes[i].get('currentTime') >= currentTimeSplit or currentTimeSplit == (timeSplit * 5):
+            j += 1
+            if self.notes[i].get('currentTime') >= currentTimeSplit or i == len(self.notes) - 1:
                 averageMIDI.append(round(average / j, 2))
-                currentTimeSplit = currentTimeSplit + timeSplit
+                currentTimeSplit += timeSplit
                 j = 0
                 average = 0
         return averageMIDI
```

Comment:
You might be better off using `pandas.DataFrame` and `pandas.Series` objects and let the lib do the calculations for you